lookout/style/typos/cmdline_tools.py

```diff
@@ -0,0 +1,180 @@
+
+    # General options
+    slogging.add_logging_args(parser)
+    subparsers = parser.add_subparsers(help="Commands")
+
+    def add_parser(name, help):
+        return subparsers.add_parser(
+            name, help=help, formatter_class=ArgumentDefaultsHelpFormatterNoNone)
+
+    # Prepare raw data for corrector
+    prepare_parser = add_parser("prepare-data", "Prepare raw dataset for corrector training.")
+    prepare_parser.set_defaults(handler=prepare_data)
+    add_preparation_config_arg(prepare_parser)
+
+    # Train new fasttext model on gien data
+    fasttext_parser = add_parser("train-fasttext", "Train fasttext model on the given dataset"
+                                                   "of code identifiers.")
+    fasttext_parser.set_defaults(handler=cli_train_fasttext)
+    add_data_path_arg(fasttext_parser)
+    add_fasttext_config_arg(fasttext_parser)
+
+    # Create train and test datasets with artificial typos
+    datasets_parser = add_parser("get-datasets",
+                                 "Create the train and the test datasets of typos.")
+    datasets_parser.set_defaults(handler=cli_get_datasets)
+    add_data_path_arg(datasets_parser)
+    add_datasets_config_arg(datasets_parser)
+
+    # Create, train and evaluate new corrector model
+    train_parser = add_parser("train", "Create and train TyposCorrector model on the given data.")
+    train_parser.set_defaults(handler=cli_train_corrector)
+    add_data_path_arg(train_parser)
+    train_parser.add_argument(
+        "-t", "--test-data-path", required=True, type=str,
+        help=".csv dump of a Dataframe with columns Columns.Split and Columns.Frequency."
+    )
+    train_parser.add_argument(
+        "-v", "--vocabulary-path", required=True, type=str,
+        help="Path to a .csv file with vocabulary."
+    )
+    train_parser.add_argument(
+        "-f", "--frequencies-path", required=True, type=str,
+        help="Path to a .csv file with tokens' frequencies."
+    )
+    train_parser.add_argument(
+        "-e", "--fasttext-path", required=True, type=str,
+        help="Path to a FastText model's dump (.bin)."
+    )
+    add_save_model_path_arg(train_parser)
+
+    # Create and train new corrector model from scratch
+    train_from_scratch_parser = add_parser(
+        "train-from-scratch", "Create and train TyposCorrector  model on the given data.")
+    train_from_scratch_parser.set_defaults(handler=train_from_scratch)
+    add_preparation_config_arg(train_from_scratch_parser)
+    add_fasttext_config_arg(train_from_scratch_parser)
+    add_datasets_config_arg(train_from_scratch_parser)
+    train_from_scratch_parser.add_argument(
+        "-e", "--fasttext-path", type=str, required=False,
+        help="Path to the pretrained fasttext model. If not specified correctly, "
+             "new fasttext model will not be trained."
+    )
+    add_save_model_path_arg(train_from_scratch_parser)
```

Comment:
I miss one last command here: `do_everything_in_one_single_command_without_parameters` 😄

Reply:
Hm, this actually must be done right here, in `train_from_scratch`, just by using default for `save_model_path`:)

Reply:
Ah OK. Can you please mark it with

```python
########################################
# single command to rule them all
########################################
```

so that nobody misses it :)